src/shared/utils/courses.ts

```diff
@@ -0,0 +1,25 @@
+import { Section } from '../fetchers';
+
+type sectionType = 'lecture' | 'lab' | 'tutorial';
+
+export const getFirstSectionType = (sections: Section[], type: sectionType): number => {
+  let i = 0;
+  for (const section of sections) {
+    if (section.sectionType === type) {
+      return i;
+    }
+    i++;
+  }
+
+  return -1;
```

Comment:
I feel this can be replace with a tertiary operator + https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex